services/nexus/nexus.service.js

```diff
@@ -1,130 +1,230 @@
+          host: 'repository.jboss.org/nexus',
+          groupId: 'ai.h2o',
+          artifactId: 'h2o-automl',
+        },
+        staticExample: this.render({
+          version: '3.22.0.2',
+        }),
+      },
+      {
+        title: 'Sonatype Nexus (Query Options)',
+        pattern: ':repo/:scheme/:host/:groupId/:artifactId/:queryOpt',
+        namedParams: {
+          repo: 'fs-public-snapshots',
+          scheme: 'https',
+          host: 'repository.jboss.org/nexus',
+          groupId: 'com.progress.fuse',
+          artifactId: 'fusehq',
+          queryOpt: ':c=agent-apple-osx:p=tar.gz',
+        },
+        staticExample: this.render({
+          version: '7.0.1-SNAPSHOT',
+        }),
+        documentation: `
+        <p>
+          Note that you can use query options with any Nexus badge type (Releases, Snapshots, or Repository)
+        </p>
+        <p>
+          Query options should be provided as key=value pairs separated by a semicolon
+        </p>
+        `,
       },
     ]
   }
 
-  static registerLegacyRouteHandler({ camp, cache }) {
-    // standalone sonatype nexus installation
+  transform({ repo, json }) {
+    let version
+    if (repo === 'r') {
+      version = json.data[0].latestRelease
+    } else if (repo === 's') {
+      for (const artifact of json.data) {
+        if (isSnapshotVersion(artifact.latestSnapshot)) {
+          version = artifact.latestSnapshot
+          break
+        }
+        if (isSnapshotVersion(artifact.version)) {
+          version = artifact.version
+          break
+        }
+      }
+    } else {
+      version = json.data.baseVersion || json.data.version
+    }
+
+    if (!version) {
+      throw new InvalidResponse({ prettyMessage: 'invalid artifact version' })
+    }
+
+    return {
+      version,
+    }
+  }
+
+  async handle({ repo, scheme, host, groupId, artifactId, queryOpt }) {
+    const requestParams = this.getRequestParams({
+      repo,
+      scheme,
+      host,
+      groupId,
+      artifactId,
+      queryOpt,
+    })
+    const json = await this._requestJson(requestParams)
+    if (json.data.length === 0) {
+      throw new NotFound({ prettyMessage: 'artifact or version not found' })
+    }
+    const { version } = this.transform({ repo, json })
+    return this.constructor.render({ version })
+  }
+
+  addQueryParamsToQueryString({ qs, queryOpt }) {
+    // Users specify query options with 'key=value' pairs, using a
+    // semicolon delimiter between pairs ([:k1=v1[:k2=v2[...]]]).
+    // queryOpt will be a string containing those key/value pairs,
+    // For example:  :c=agent-apple-osx:p=tar.gz
+    const keyValuePairs = queryOpt.split(':')
+    keyValuePairs.forEach(keyValuePair => {
+      const paramParts = keyValuePair.split('=')
+      const paramKey = paramParts[0]
+      const paramValue = paramParts[1]
+      qs[paramKey] = paramValue
+    })
+  }
+
+  getRequestParams({ repo, scheme, host, groupId, artifactId, queryOpt }) {
```

Comment:
This probably would be a bit more direct if it invoked `_requestJson` and was renamed to `fetch`. "Fetch" is another pattern we use a lot.